### DISCUSSION TITLE
Uniaxial stretching simulation with coarse mesh or fine mesh based on J2 plasticity

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#19390] [Q&A Modules: Solid mechanics] [answered]
Dear MOOSE experts,

Recently, I conducted two uniaxial stretching simulations based on J2 plasticity. The first one (test1) has a relatively coarse mesh. Both the stretching curve and the calculation running time are very nice, as shown in Fig.1. However, the second one (test2) has a finer grid , and the calculated curve is correct, but `dt` gets smaller and smaller after 10s as the simulation progresses, until it reaches the set `dtmin = 2e-6 `, the final solution failed, as shown in Figure 2.  My input file is [j2plastic_1.i](https://github.com/PengWei97/monkey/blob/main/problem/J2/coarse-fine_mesh/j2plastic_1.i)

![image](https://user-images.githubusercontent.com/56183775/141799356-ae3555c6-c152-461c-9804-ce7422a6ad37.png)
![image](https://user-images.githubusercontent.com/56183775/141799355-883cd341-5df9-467e-9f1b-db06e034c26b.png)

So, my question is what is the root cause of this bug? And I want to emphasize that the two tests are only different in the number of grids. Finally, what aspects should be taken to solve this problem?

Any suggestions or recommendations to fix the problem would be greatly appreciated.

Thank you
Wei

**Comment by jiangwen84** [comment, marked as answer]
Can you try a different solver? 
```
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
```

**Reply by PengWei97** [reply]
This problem seems to be resolved, and the result is shown in Figure 3.
![image](https://user-images.githubusercontent.com/56183775/141887501-11229aa6-e17a-4949-8b5d-b2729cd5acfe.png)

Although I don't know the difference between `lu` and `hypre`. Regarding mechanical problems, whether it is J2 or crystal plasticity, do you have any good suggestions and experience to share with regard to the choice of solver?

**Reply by jiangwen84** [reply]
They are different linear solvers. LU is a direct solver. The direct solvers are generally more robust than iterative solver, but will use more memory. If your problem size is not very large, LU is recommended for plasticity problems.

**Reply by PengWei97** [reply]
Ok, I got it. Thank again for your advice.
